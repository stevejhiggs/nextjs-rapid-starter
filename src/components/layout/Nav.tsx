import React from 'react';
import Link from 'next/link';

interface NavElement {
  url: string;
  text: string;
}

const navUrls: NavElement[] = [
  {
    url: '/',
    text: 'Home'
  },
  {
    url: '/about',
    text: 'About'
  },
  {
    url: '/customComponents',
    text: 'Custom components'
  },
  {
    url: '/serverRendered',
    text: 'Server rendered'
  },
  {
    url: '/staticallyGenerated',
    text: 'Statically generated'
  },
  {
    url: '/showRouter/blahblahMadeUpValue',
    text: 'Dynamic route'
  },
  {
    url: '/i18n/staticPage',
    text: 'i18n - Static'
  },
  {
    url: '/i18n/serverRendered',
    text: 'i18n - Server rendered'
  },
  {
    url: '/getStaticPaths/blogArticleA',
    text: 'GetStaticPaths'
  },
  {
    url: '/getStaticPathsChildren/blog/articles/A',
    text: 'GetStaticPaths - Nested'
  }
];

function mapLinkCollection(links: NavElement[]) {
  return links.map((navElement) => (
    <li key={navElement.url} className="pb-1 px-1">
      <Link href={navElement.url}>{navElement.text}</Link>
    </li>
  ));
}

const Nav: React.FC = () => <ul>{mapLinkCollection(navUrls)}</ul>;

export default Nav;
