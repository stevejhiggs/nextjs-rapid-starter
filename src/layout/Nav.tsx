import React from 'react';
import Link from 'next/link';
import { Box, Link as ChakraLink } from '@chakra-ui/react';

interface NavElement {
  url: string;
  text: string;
}

const navUrls: NavElement[] = [
  {
    url: '/',
    text: 'Home'
  },
  {
    url: '/about',
    text: 'About'
  },
  {
    url: '/customComponents',
    text: 'Custom components'
  },
  {
    url: '/serverRendered',
    text: 'Server rendered'
  },
  {
    url: '/staticallyGenerated',
    text: 'Statically generated'
  },
  {
    url: '/showRouter/blahblahMadeUpValue',
    text: 'Dynamic route'
  },
  {
    url: '/i18n/staticPage',
    text: 'i18n - Static'
  },
  {
    url: '/i18n/serverRendered',
    text: 'i18n - Server rendered'
  },
  {
    url: '/getStaticPaths/blogArticleA',
    text: 'GetStaticPaths'
  },
  {
    url: '/getStaticPathsChildren/blog/articles/A',
    text: 'GetStaticPaths - Nested'
  }
];

function mapLinkCollection(links: NavElement[]) {
  return links.map((navElement) => (
    <Box as="li" key={navElement.url} paddingBottom="0.5rem" paddingLeft="0.5rem">
      <Link href={navElement.url} passHref>
        <ChakraLink>{navElement.text}</ChakraLink>
      </Link>
    </Box>
  ));
}

const Nav: React.FC = () => <Box as="ul">{mapLinkCollection(navUrls)}</Box>;

export default Nav;
