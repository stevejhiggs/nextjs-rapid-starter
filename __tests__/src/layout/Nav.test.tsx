import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Nav from '@/src/layout/Nav';

afterEach(cleanup);

describe('Components -> Nav', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Nav />);
    expect(container).toMatchSnapshot();
  });
});
