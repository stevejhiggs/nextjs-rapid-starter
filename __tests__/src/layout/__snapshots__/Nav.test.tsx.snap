// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components -> Nav should match the snapshot 1`] = `
<div>
  <ul
    class="css-0"
  >
    <li
      class="css-pmguu6"
    >
      <a
        class="chakra-link css-0"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="css-pmguu6"
    >
      <a
        class="chakra-link css-0"
        href="/about"
      >
        About
      </a>
    </li>
    <li
      class="css-pmguu6"
    >
      <a
        class="chakra-link css-0"
        href="/customComponents"
      >
        Custom components
      </a>
    </li>
    <li
      class="css-pmguu6"
    >
      <a
        class="chakra-link css-0"
        href="/serverRendered"
      >
        Server rendered
      </a>
    </li>
    <li
      class="css-pmguu6"
    >
      <a
        class="chakra-link css-0"
        href="/staticallyGenerated"
      >
        Statically generated
      </a>
    </li>
    <li
      class="css-pmguu6"
    >
      <a
        class="chakra-link css-0"
        href="/showRouter/blahblahMadeUpValue"
      >
        Dynamic route
      </a>
    </li>
    <li
      class="css-pmguu6"
    >
      <a
        class="chakra-link css-0"
        href="/i18n/staticPage"
      >
        i18n - Static
      </a>
    </li>
    <li
      class="css-pmguu6"
    >
      <a
        class="chakra-link css-0"
        href="/i18n/serverRendered"
      >
        i18n - Server rendered
      </a>
    </li>
    <li
      class="css-pmguu6"
    >
      <a
        class="chakra-link css-0"
        href="/getStaticPaths/blogArticleA"
      >
        GetStaticPaths
      </a>
    </li>
    <li
      class="css-pmguu6"
    >
      <a
        class="chakra-link css-0"
        href="/getStaticPathsChildren/blog/articles/A"
      >
        GetStaticPaths - Nested
      </a>
    </li>
  </ul>
</div>
`;
